content/es/docs/concepts/configuration/secret.md

```diff
@@ -0,0 +1,907 @@
+es más seguro y más flexible que ponerlo en la definición de un {{< glossary_tooltip term_id="pod" >}} o en un {{< glossary_tooltip text="container image" term_id="image" >}}. Ver [Secrets design document](https://git.k8s.io/community/contributors/design-proposals/auth/secrets.md) para más información.
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Introducción a Secrets
+
+Un `Secret` es un objeto que contiene una pequeña cantidad de datos confidenciales como contraseñas, un token, o una llave.  Tal información podría ser puesta en la especificación de un `Pod` o en una imagen; poniendolo en un objeto de tipo `Secret` permite mayor control sobre como se usa, y reduce el riesgo de exposicición accidental.
+
+Los usuarios pueden crear Secrets, y el sistema también puede crearlos.
+
+Para usar un `Secret`, un `Pod` debe hacer referencia a este. Un Secret puede ser usado con un `Pod`  de dos formas: como archivos en un {{< glossary_tooltip text="volumen" term_id="volume" >}} montado en uno o más de sus contenedores, o utilizados por `kubelet` al extraer imágenes del pod.
+
+### Secrets incorporados
+
+#### Las Cuentas de Servicio Crean y Adjuntan Secrets con las Credenciales de la API
+
+Kubernetes crea automaticamente Secrets que contienen credenciales para acceder a la API y modifica automáticamente sus pods para usar este tipo de Secret.
+
+La creación y el uso automático de las credenciales de la API, pueden desabilitarse o anularse si se desea. Sin embargo, si todo lo que necesita hacer es acceder de forma segura al apiserver, este es el flujo de trabajo recomendado.
+
+Ver la documentación de  [Service Account](/docs/tasks/configure-pod-container/configure-service-account/) para más información sobre cómo funcionan las Cuentas de Servicio.
+
+### Creando tu propio Secret
+
+#### Creando un Secret Usando kubectl create Secret
+
+Digamos que algunos pods necesitan acceder a una base de datos. El nombre y contraseña que los pods deberían usar estan en los archivos:
+`./username.txt` y `./password.txt` en tu máquina local.
+
+```shell
+# Crear archivos necesarios para el resto del ejemplo.
+echo -n 'admin' > ./username.txt
+echo -n '1f2d1e2e67df' > ./password.txt
+```
+
+El comando `kubectl create secret`
+empaqueta esos archivos en un Secret y crea el objeto en el Apiserver.
+
+```shell
+kubectl create Secret generic db-user-pass --from-file=./username.txt --from-file=./password.txt
+```
+```
+Secret "db-user-pass" created
+```
+{{< note >}}
+Caracteres especiales como `$`, `\*`, y `!` requieren escapar.
+Si la contraseña que está utilizando tiene caracteres especiales, debe escapar de ellos usando el caracter `\\` . Por ejemplo, si tu password actual es `S!B\*d$zDsb`, deberías ejecutar el comando de esta manera: kubectl create Secret generic dev-db-secret --from-literal=username=devuser --from-literal=password=S\\!B\\\\*d\\$zDsb . No necesita escapar de caracteres especiales en contraseñas de archivos (`--from-file`).
```

Comment:
The formatting on this is different from the English original. I recommend copying the original Markdown layout.